File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/TodayFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class TodayFunction : NoArgumentFunction
    {
        public const string Name = "TODAY";

        protected override VariableType ResultType => PrimitiveType.Date;

        private TodayFunction(SourceReference reference)
            : base(reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference) => new TodayFunction(reference);
    }
}

File: typescipt/src/language/expressions/functions/todayFunction.ts


export class TodayFunction extends NoArgumentFunction {
   public const string Name = `TODAY`;

   protected override VariableType ResultType => PrimitiveType.date;

   constructor(reference: SourceReference)
     {
     super(reference);
   }

   public static create(reference: SourceReference): ExpressionFunction {
     return new TodayFunction(reference);
   }
}
